Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -921,20 +923,21 @@ def fetch(self, branch, remote='origin'):
 
     def pull(self, rebase=None, branch=None, remote='origin'):
         commit = self.commit() if self.is_svn or branch else None
-        code = run(
-            [self.executable(), 'pull'] + (
-                [remote, branch] if branch else []
-            ) + (
-                [] if rebase is None else ['--rebase={}'.format('True' if rebase else 'False')]
-            ), cwd=self.root_path,
-        ).returncode
-        if self.cache and rebase and branch != self.branch:
-            self.cache.clear(self.branch)
 
-        if not code and branch:
+        code = 0
+        if branch and self.branch != branch:
             code = self.fetch(branch=branch, remote=remote)
+        if not code:
+            command = [self.executable(), 'pull'] + ([remote, branch] if branch else [])
+            if rebase is True:
+                command += ['--rebase=True', '--autostash']
+            if rebase is False:
+                command += ['--rebase=False']
+            code = run(command, cwd=self.root_path).returncode
```

Comment:
We don't specify --rebase option when rebase is not set?
Doesn't this mean `git-webkit pull` wouldn't do the right thing?

Reply:
ok